splinterd/tests/framework/builders/mod.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2018-2021 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Framework of builders for integration testing
+
+mod circuit;
+mod error;
+mod scabbard;
+
+pub use self::circuit::{CircuitBuilder, CircuitData, CircuitService};
+pub use self::error::{AddScabbardServiceError, CircuitBuildError};
+pub use self::scabbard::ScabbardCircuitBuilder;
+pub use splinter::admin::messages::CreateCircuit;
+
+use crate::framework::network::Network;
+use splinter::error::InvalidArgumentError;
+use splinterd::node::Node;
+
+/// A generic network
+pub trait Networkable {
```

Comment:
Rename this something to do with getting/supplying nodes and it will make more sense--NodeCollection perhaps. There really isn't a compelling reason not to directly use Network--except to constrain the portion of the API being used by the circuit builder module (which is a good enough reason imo), so this should purely be named from circuit builder's pov.